openstates/va/bills.py

```diff
@@ -1,90 +1,109 @@
-        ('Read third time and defeated', 'bill:failed'),
-        ('Presented', 'bill:introduced'),
-        ('Prefiled and ordered printed', 'bill:introduced'),
-        ('Read first time', 'bill:reading:1'),
-        ('Read second time', 'bill:reading:2'),
-        ('Read third time', 'bill:reading:3'),
-        ('Senators: ', None),
-        ('Delegates: ', None),
-        ('Committee substitute printed', None),
-        ('Bill text as passed', None),
-        ('Acts of Assembly', None),
-    )
-
-    link_xpath = '//ul[@class="linkSect"]/li/a'
-
-    def _accept_response(self, response):
-        # check for rate limit pages
-        normal = super(VABillScraper, self)._accept_response(response)
-        return (normal and
-            'Sorry, your query could not be processed' not in response.text
-            and 'the source database is temporarily unavailable' not in response.text)
-
-    def _get_page_bills(self, issue_name, href):
-        issue_html = self.get('http://lis.virginia.gov' + href,
-                                  retry_on_404=True).text
-        idoc = lxml.html.fromstring(issue_html)
-        for ilink in idoc.xpath(self.link_xpath):
-            self.subject_map[ilink.text].append(issue_name)
-
-        more_links = idoc.xpath('//a/b[text()="More..."]/../@href')
-        if more_links:
-            self._get_page_bills(issue_name, more_links[0])
-
-    def _build_subject_map(self):
-        url = 'http://lis.virginia.gov/cgi-bin/legp604.exe?%s+sbj+SBJ' % self.site_id
-        self.subject_map = defaultdict(list)
-
-        # loop over list of all issue pages
-        html = self.get(url).text
-        doc = lxml.html.fromstring(html)
-        for link in doc.xpath(self.link_xpath):
-            # get bills from page
-            self._get_page_bills(link.text, link.get('href'))
-
-    def _fetch_sponsors(self, bill):
-        url = "http://lis.virginia.gov/cgi-bin/legp604.exe?%s+mbr+%s" % (
-            self.site_id, bill['bill_id'].replace(' ', ''))
-
-        html = self.get(url, retry_on_404=True).text
-        doc = lxml.html.fromstring(html)
-
-        for slist in doc.xpath('//ul[@class="linkSect"]'):
+            # create a bill
+            desc = bill.xpath('text()')[0].strip()
+            chamber = {
+                'H': 'lower',
+                'S': 'upper',
+            }[bill_id[0]]
+            bill_type = {
+                'B': 'bill',
+                'J': 'joint resolution',
+                'R': 'resolution'
+            }[bill_id[1]]
+            bill = Bill(bill_id, self.kwargs['session'], desc,
+                        chamber=chamber, classification=bill_type)
+
+            bill_url = link.get('href')
+            sponsor_url = BASE_URL + URL_PATTERNS['sponsors'].format(
+                self.kwargs['session'],
+                bill_id.replace(' ', ''),
+            )
+
+            list(self.scrape_page_items(BillSponsorPage, url=sponsor_url, obj=bill))
```

Comment:
I see why this is going on but these could probably use a comment just so it is clear what is going on here  (and/or we should come up with a less hacky way for spatula to handle this case)